app/routes.py

```diff
@@ -1 +1,137 @@
+    return jsonify(response), 200
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_id(Task, task_id)
+    return make_response({"task": task.to_dict()}, 200)
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = validate_id(Task, task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+    return make_response({"task": task.to_dict()}, 200)
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_id(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+    return make_response({"details": f"Task {task.id} \"{task.title}\" successfully deleted"}, 200)
+
+@task_bp.route("/<task_id>/<complete_status>", methods=["PATCH"])
+def mark_complete(task_id, complete_status):
+    task = validate_id(Task, task_id)
+    if complete_status == "mark_complete":
+        task.completed_at = datetime.now()
+        requests.post("https://slack.com/api/chat.postMessage", json={"channel": "task-notifications", "text": f"Someone just completed the task {task.title}"}, headers={"Authorization": os.environ.get("SLACK_BOT_TOKEN")})
+    elif complete_status == "mark_incomplete":
+        task.completed_at = None
+
+    db.session.commit()
+    return make_response({"task": task.to_dict()}, 200)
+
+goal_bp = Blueprint("goal", __name__, url_prefix="/goals")
+@goal_bp.route("", methods=["POST"])
+def create_one_goal():
+    request_body = request.get_json()
+    new_goal = validate_item(Goal, request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+    return make_response({"goal": new_goal.to_dict()}, 201)
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    response = [goal.to_dict() for goal in goals]
+    return jsonify(response), 200
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_id(Goal, goal_id)
+    return make_response({"goal": goal.to_dict()}, 200)
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_one_goal(goal_id):
+    goal = validate_id(Goal, goal_id)
+    request_body = request.get_json()
+    goal.title = request_body["title"]
+
+    db.session.commit()
+    return make_response({"goal": goal.to_dict()}, 200)
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_one_goal(goal_id):
+    goal = validate_id(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+    return make_response({"details": f"Goal {goal.id} \"{goal.title}\" successfully deleted"}, 200)
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_tasks_to_goal(goal_id):
+    goal = validate_id(Goal, goal_id)
+    request_body = request.get_json()
+    task_ids = request_body["task_ids"]
+    goal.tasks = [Task.query.get(id) for id in task_ids]
```

Comment:
If there were already tasks attached to the goal, those would get removed. It would be better to append to the existing goal tasks list.